### PR TITLE
Enable aligning a dropdown button's width with its menu's width 

### LINKED ISSUE #14785: DropdownMenuItem widget is wider than the Dropdown itself
## Steps to Reproduce
Copied from https://stackoverflow.com/questions/48895195/how-can-we-change-the-width-padding-of-a-flutter-dropdownmenuitem-in-a-dropdown

In Flutter, I can build a Dropdown with DropdownMenuItems, like this:
[![enter image description here][1]][1]

The DropdownMenuItems I add are always wider than the dropdown itself:

[![enter image description here][2]][2]

How do you adjust the width of the DropdownMenuItem, or remove the extra horizontal padding?

My DropdownMenuItem widget looks like this:

    DropdownMenuItem(
        value: unit.name,
        child: Text('hey'),
    );

while my Dropdown widget looks like this:

    return Container(
        width: 300.0,
        child: DropdownButtonHideUnderline(
          child: DropdownButton(
            value: name,
            items: listOfDropdownMenuItems,
            onChanged: onChanged,
            style: Theme.of(context).textTheme.title,
          ),
        ),
    );

  [1]: https://i.stack.imgur.com/GnwfL.png
  [2]: https://i.stack.imgur.com/D5LI1.png

Goal is to just have the same width between the Dropdown and the DropdownMenuItems; we don't need to make the width actually adjustable.

### PR DESCRIPTION
**Pull request by HansMuller**
By default dropdown menus are wider than their dropdown button: 16.0 pixels wider on the left, and 24.0 pixels wider on the right.

Setting `ButtonThemeData.alignedDropdown` to true makes dropdown menus the same width as their button. The button's label is padded 16.0 on the left so that the button and menu item labels line up.

```dart
new ButtonTheme(
  alignedDropdown: true,
  child: new DropdownButton( ... ),
)
```

Fixes https://github.com/flutter/flutter/issues/14785

### CODE REVIEW

File: packages/flutter/lib/src/material/button_theme.dart [hunk 1]
```diff
@@ -229,6 +237,15 @@ class ButtonThemeData extends Diagnosticable {
...
   final ShapeBorder _shape;
 
+  /// If true, then a [DropdownButton] menu's width will match the button's
+  /// width.
+  ///
+  /// If false (the default), then the dropdown's menu will be wider than
+  /// its button. In either case the dropdown button will line up the leading
+  /// edge of the menu's value with the leading edge of the values
+  /// displayed by the menu items.
+  final bool alignedDropdown;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/button_theme.dart]: maybe add something like "has no effect on buttons other than [DropdownButton]" or something?

**Comment by Hixie** [comment]
LGTM if you decide this is the right solution.

**Comment by HansMuller** [comment]
I do think this is the right solution. It's backwards compatible, since by default dropdown menu geometry doesn't change. And it makes it simple to enable the new and improved menu geometry with the material theme's button theme.

**Comment by jhionan** [comment]
Thank you for this PR, but next time please use a nicer commit messages, checkpoint is not a commit!

**Pull request merged**